### DISCUSSION TITLE
Suggestion: testClient.deal

### DISCUSSION THREAD

**Discussion by marcelomorgado** [wevm/viem#2259] [Idea / Feature Request]
It would be a handy function to set token balances

**Comment by tmm** [comment]
PR welcome

**Reply by marcelomorgado** [reply]
Hi, unlike `Foundry` - that has a way more powerful storage handle utils -, we would have to keep a mapping with chains, tokens and their slots. This means that for every new token we want to support for the `deal` function a PR will be needed.
If we add a big list from begging it won't be an often issue but I think worth pointing it.
That all said, is it worth going forward and implementing this?

Refs: https://github.com/foundry-rs/forge-std/blob/52715a217dc51d0de15877878ab8213f6cbbbab5/src/StdCheats.sol#L728-L747

**Comment by marcelomorgado** [comment]
Duplicated https://github.com/wevm/viem/discussions/2819
